File: app/src/main/java/com/antyzero/smoksmog/time/CountdownProvider.java
package com.antyzero.smoksmog.time;


import android.content.Context;

import java.util.Locale;

public class CountdownProvider {

    private static final Locale LOCALE_POLISH = new Locale("pl", "PL");

    private final Context context;

    public CountdownProvider(Context context) {
        this.context = context;
    }

    public String get(int seconds) {

        Locale locale = context.getResources().getConfiguration().locale;

        if (LOCALE_POLISH.equals(locale)) {
            return new PolishCountdown().get(seconds);
        } else {
            return new EnglishCountdown().get(seconds);
        }
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/time/CountdownProvider.kt
package com.antyzero.smoksmog.time


import android.content.Context
import com.antyzero.smoksmog.dsl.getCompatLocale
import java.util.*

class CountdownProvider(private val context: Context) {

    operator fun get(seconds: Int): String = when (context.resources.configuration.getCompatLocale()) {
        LOCALE_POLISH -> PolishCountdown()[seconds]
        else -> EnglishCountdown()[seconds]
    }

    companion object {

        private val LOCALE_POLISH = Locale("pl", "PL")
    }
}
